### PR TITLE
chore: document the development experience for a component

### LINKED ISSUE #30: chore(tests): add unit tests documentation
* Add documentation on how to run the tests. 
* Add documentation on how to develop with the tests
* Add a few notes on how it was developed (base configuration and spec configuration etc.).

### PR DESCRIPTION
**Pull request by YonatanKra**
Closes #30

### CODE REVIEW

File: docs/development-process.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Development Process
+Please watch [the following video](https://drive.google.com/file/d/1pvSKmGAm-uoIfKyZBarrapwm5q_wzAs5/view) in order to understand the development process.
```
Comment (yinonov) [inline docs/development-process.md:2]: paste the video right here?

File: docs/development-process.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+## Tl;Dw
+1. Run `npm test:dev` to run the tests in watch mode
+2. In order to isolate a test, set its `describe` to `fdescribe` or even a specific `it` to `fit`
+3. From now on, every change you'll make will be reflected in the tests
+4. In order to debug visually, 
+   1. Make sure to that the `afterEach` that clears the DOM from the tests' artifacts is not active.
```
Comment (yinonov) [inline docs/development-process.md:9]:
```suggestion
   1. Make sure that the `afterEach` that clears the DOM from the tests' artifacts is not active.
```

Comment (yinonov) [reply docs/development-process.md:9]: maybe rephrase that?
Ensure `afterEach` method (that clears the DOM from the tests' artifacts) is inactive

File: docs/development-process.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+   2. Click on the "Debug" button at the top of the test browser
+5. In webstorm, use `cmd` + `shift` + `t` in order to toggle between the test and the business logic file.
```
Comment (yinonov) [inline docs/development-process.md:11]: webstorm related? u can toggle in vscode as well with cmd + p. important here?

File: karma.conf.ts [hunk 4]
```diff
@@ -14,6 +14,10 @@ export default function (config: VividKarmaConfig) {
             {
                 pattern: 'src/**/*.spec.ts',
                 watched: false
+            },
+            {
+                pattern: 'src/core/theme/**/*.scss',
+                watched: true
```
Comment (yinonov) [inline karma.conf.ts:20]: we still need to decide if this is a coupled API or a "3rd party asset" what if we have more assets to observe?

File: karma.conf.ts [hunk 5]
```diff
@@ -25,6 +29,7 @@ export default function (config: VividKarmaConfig) {
         concurrency: Infinity,
         preprocessors: {
             'src/**/*.spec.ts': ["rollup"],
+            'src/core/theme/**/*.scss': ['scss']
```
Comment (yinonov) [inline karma.conf.ts:32]: (same)

File: src/components/badge/badge.scss [hunk 6]
```diff
@@ -1,63 +1,121 @@
...
-}
-.icon>vwc-icon{
-    --icon-size: 12px;--connotation:initial;vertical-align:top
-}
-.icon.icon--leading{
-    margin-inline-end:8px
-}
-.icon.icon--trailing{
-    margin-inline-start:8px
+.badge.layout-filled, .badge:not(.layout-outlined):not(.layout-soft) {
```
Comment (yinonov) [inline src/components/badge/badge.scss:1]: I wouldn't do that just yet. we'll have stylelint to do that for us

File: package.json [hunk 7]
```diff
@@ -85,6 +85,8 @@
     "karma-jasmine": "^4.0.1",
     "karma-rollup-preprocessor": "^7.0.7",
     "karma-safarinative-launcher": "^1.1.0",
+    "karma-scss-preprocessor": "^4.0.0",
+    "node-sass": "^6.0.1",
```
Comment (yinonov) [inline package.json:89]: where do we use this?

**Review by yinonov — CHANGES_REQUESTED** [review]

File: docs/development-process.md (hunk 3 above)
Comment (YonatanKra) [reply docs/development-process.md:11]: that's different. cmd + p is to search for files. In webstorm you toggle between code and its test file.

File: karma.conf.ts (hunk 4 above)
Comment (YonatanKra) [reply karma.conf.ts:20]: If we have more assets, we will bring them here. Right now this is the setup :)

File: karma.conf.ts (hunk 5 above)
Comment (YonatanKra) [reply karma.conf.ts:32]: (same) :)

File: package.json (hunk 7 above)
Comment (YonatanKra) [reply package.json:89]: Inside karma

File: docs/development-process.md (hunk 1 above)
Comment (YonatanKra) [reply docs/development-process.md:2]: Care to show me how?
The only way I saw was to add an image that is a link to the video.

Comment (yinonov) [reply docs/development-process.md:2]: edit in GitHub
image

File: docs/development-process.md (hunk 3 above)
Comment (yinonov) [reply docs/development-process.md:11]: toggle between symbols in page is also available. I just think it's an IDE related
if must, add it as a small caption hint/tip?
`CMD` + `SHIFT` + `O`
image

File: package.json (hunk 7 above)
Comment (yinonov) [reply package.json:89]: why not use the sass that is already a dependency? is it Karma's peer d.?

Comment (YonatanKra) [reply package.json:89]: It's the karma processor peer. I could write my own karma-sass processor, but thought to take this shortcut. It is a peer dependency of the karma-sass-processor package I'm using here.

**Review by yinonov — APPROVED** [review]
I still believe the 5th point is a little off
